src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -104,6 +93,34 @@ public static class ExpressionMetadataProvider
                 metadata = metadataProvider.GetMetadataForType(containerType).Properties[propertyName];
             }
 
+            Func<object, object> modelAccessor = (container) =>
```

Comment:
I'm very hesitant about making any changes to this `Func`. It's just too general. For example, your changes here will affect how `enum`s are displayed in `<input/>` tag helpers. Think we want something specific to `@Html.Display[For]()`, `@Html.DropDownList[For]()`, `@Html.ListBox[For]()`, and the `<select>` tag helper.